src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -103,12 +103,16 @@ async def producer(
         self,
         inputs: Sequence[Any],
         queue: asyncio.Queue[Tuple[int, Any]],
+        max_fill: int,
         done_producing: asyncio.Event,
     ) -> None:
         try:
             for index, input in enumerate(inputs):
                 if self._TERMINATE.is_set():
                     break
+                while queue.qsize() >= max_fill:
+                    # keep room in the queue for requeues
+                    await asyncio.sleep(1)
```

Comment:
If I am understanding correctly, the max queue size is currently twice the concurrency and the max fill is 4 times the concurrency, so will this condition ever be satisfied?

Reply:
ah yes I meant to pass in the new max_queue_size, thanks